e2e-tests/tests/test_jolteon_consensus_rpc.py

```diff
@@ -0,0 +1,424 @@
+        that the Jolteon consensus is making progress.
+        """
+        logger.info("Starting Jolteon round progression test")
+        
+        # Get initial round state
+        initial_result = api.substrate.rpc_request("jolteon_getRoundInfo", [])
+        if not initial_result or 'result' not in initial_result:
+            assert False, "Failed to get initial round info"
+        
+        initial_round = initial_result['result']
+        logger.info(f"Initial round state: {initial_round}")
+        
+        # Wait for round progression
+        wait_time = 30  # Wait 30 seconds for round advancement
+        logger.info(f"Waiting {wait_time} seconds for round progression...")
+        sleep(wait_time)
+        
+        # Get final round state
+        final_result = api.substrate.rpc_request("jolteon_getRoundInfo", [])
+        if not final_result or 'result' not in final_result:
+            assert False, "Failed to get final round info"
+        
+        final_round = final_result['result']
+        logger.info(f"Final round state: {final_round}")
+        
+        # Verify round progression
+        assert final_round['r_curr'] >= initial_round['r_curr'], \
+            f"Current round should not decrease: {initial_round['r_curr']} -> {final_round['r_curr']}"
+        
+        # In happy path, rounds should advance
+        if final_round['r_curr'] > initial_round['r_curr']:
+            logger.info(f"✅ Rounds advanced: {initial_round['r_curr']} -> {final_round['r_curr']}")
+        else:
+            logger.info(f"⚠️  Rounds stayed the same: {initial_round['r_curr']}")
+        
+        # Verify round consistency
+        assert final_round['r_vote'] >= initial_round['r_vote'], \
+            f"Voted round should not decrease: {initial_round['r_vote']} -> {final_round['r_vote']}"
+        
+        assert final_round['r_lock'] >= initial_round['r_lock'], \
+            f"Locked round should not decrease: {initial_round['r_lock']} -> {final_round['r_lock']}"
+        
+        logger.info("✅ Round progression test passed")
+
+    @mark.test_key('JOLTEON-RPC-003')
+    def test_quorum_certificate_formation(self, api: BlockchainApi, config: ApiConfig):
+        """Test Quorum Certificate formation and progression
+        
+        This test verifies that QCs are being formed correctly
+        and that the highest QC is advancing.
+        """
+        logger.info("Starting Jolteon Quorum Certificate formation test")
+        
+        # Get initial QC state
+        initial_result = api.substrate.rpc_request("jolteon_getHighestQC", [])
+        if not initial_result or 'result' not in initial_result:
+            assert False, "Failed to get initial QC"
+        
+        initial_qc = initial_result['result']
+        logger.info(f"Initial QC: {initial_qc}")
+        
+        # Wait for QC progression
+        wait_time = 45  # Wait 45 seconds for QC advancement
```

Comment:
Also here. Not sure why you wait 45 seconds, you could wait round_timeout + 1 or something like that.